tests/test_locality_NeighborQuery.py

```diff
@@ -39,20 +39,23 @@ def build_query_object(cls, box, ref_points, r_max=None):
             "subclass of NeighborQuery in a separate test subclass."
         )
 
-    def test_query_validate_points(self):
+    @pytest.mark.parametrize(
+        "empty_points",
+        [
+            [],
+            [[]],
+            np.zeros(0, dtype=np.float32),
+            np.zeros(shape=(0, 3), dtype=np.float32),
+        ],
+    )
+    def test_query_validate_points(self, empty_points):
         L = 10  # Box Dimensions
         r_max = 2.01  # Cutoff radius
         box = freud.box.Box.cube(L)
 
         # It's not allowed to have an empty NeighborQuery
```

Comment:
Only part of this test is parametrized. But the rest of the test does not depend on the values of `empty_points` and is now being tested redundantly. Either separate the parametrized portion of the test into a new test named `test_query_empty_points`, or revert the changes and keep the inner `for` loop rather than parametrizing it. I vote for breaking up the test, because this is really testing multiple kinds of behavior with a single test name.

Reply:
I also vote for 2 tests, one which tests a set of valid inputs, and another test which tests a set of invalid inputs.